server/lib/processDonation.js

```diff
@@ -0,0 +1,134 @@
+const gateways = require('../gateways');
+const constants = require('../constants/transactions');
+const roles = require('../constants/roles');
+const emailLib = require('./email');
+
+module.exports = (Sequelize, donation) => {
+
+  const services = {
+
+    stripe: (donation) => {
```

Comment:
This file is hard to read.
I wonder if it would be easier to create a new file `stripe.js` with all this and exposing a few methods that don't depend on the model. That will be easier to create other providers.